src/Examine/SearchResultsBase.cs

```diff
@@ -28,7 +28,7 @@ public static ISearchResults Empty()
         /// Gets the total number of results for the search
         /// </summary>
         /// <value>The total items from the search.</value>
-        public long TotalItemCount { get; internal set; }
+        public long TotalItemCount { get; set; }
```

Comment:
Why is the setter made public? This would allow external code to alter the total item count by casting the `ISearchResults` to this base class and using the setter... Making this a protected setter would solve this:
```suggestion
        public long TotalItemCount { get; protected set; }
```

If this property must be set from other (non-inheriting) classes, an internal setter function could be added to this base class, although that would indicate a rather bad implementation/design.

Reply:
I think this readonly getter is what is exposed on the interface, will need to review more in depth , thanks for pointing this on out!

Reply:
Yes, the interface only has the getter, so that indicates this property shouldn't be publically settable. Because the total item could is set outside of the constructor, this needs a setter (or backing field), which should be protected, as it's a base class 👍

Reply:
have made this change, thanks!